octree/include/pcl/octree/octree_iterator.h

```diff
@@ -173,8 +173,18 @@ class OctreeIteratorBase {
   inline const OctreeKey&
   getCurrentOctreeKey() const
   {
-    assert(octree_ != 0);
-    assert(current_state_ != 0);
```

Comment:
Actually following code is compilable:
```cpp
pcl::octree::OctreeIteratorBase<pcl::octree::OctreeBase<int>> octreeIter;
pcl::octree::OctreeKey octreeKey = octreeIter.getCurrentOctreeKey();
```
And it will throw "Access violation reading location" on second statement.
From the comment of `OctreeIteratorBase`, it seems it should be a abstract class but actually it's not. Maybe we should make it an abstract class?

Reply:
Seems reasonable to make it an abstract class (though one or two tests might have to be rewritten). Which function would you make pure virtual? The increment operators?